src/connectors/otel/id.rs

```diff
@@ -0,0 +1,296 @@
+// Copyright 2020-2021, The Tremor Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::errors::Result;
+use rand::Rng;
+use simd_json::StaticNode;
+use tremor_script::Value;
+
+pub(crate) fn random_span_id_bytes() -> Vec<u8> {
+    let mut rng = rand::thread_rng();
+    let span_id: Vec<u8> = (0..8).map(|_| rng.gen_range(0_u8..=255_u8)).collect();
+    span_id
+}
+
+pub(crate) fn random_span_id_string() -> String {
+    let mut rng = rand::thread_rng();
+    let span_id: String = (0..8)
+        .map(|_| rng.gen_range(0_u8..=255_u8))
+        .map(|b| format!("{:02x}", b))
+        .collect();
+    span_id
+}
+
+pub(crate) fn random_span_id_array<'event>() -> Value<'event> {
+    Value::Array(
+        random_span_id_bytes()
+            .iter()
+            .map(|b| Value::Static(StaticNode::U64(u64::from(*b))))
+            .collect(),
+    )
+}
+
+pub(crate) fn random_span_id_value<'event>() -> Value<'event> {
+    Value::String(random_span_id_string().into())
+}
+
+pub(crate) fn random_trace_id_bytes() -> Vec<u8> {
+    let mut rng = rand::thread_rng();
+    let span_id: Vec<u8> = (0..16).map(|_| rng.gen_range(0_u8..=255_u8)).collect();
+    span_id
+}
+
+pub(crate) fn random_trace_id_string() -> String {
+    let mut rng = rand::thread_rng();
+    let span_id: String = (0..16)
+        .map(|_| rng.gen_range(0_u8..=255_u8))
+        .map(|b| format!("{:02x}", b))
+        .collect();
+    span_id
+}
+
+pub(crate) fn random_trace_id_value<'event>() -> Value<'event> {
+    Value::String(random_trace_id_string().into())
+}
+
+fn hex_to_bytes(str_bytes: &[u8]) -> Vec<u8> {
+    let mut hex_bytes = str_bytes
+        .iter()
+        .filter_map(|b| match b {
+            b'0'..=b'9' => Some(b - b'0'),
+            b'a'..=b'f' => Some(b - b'a' + 10),
+            b'A'..=b'F' => Some(b - b'A' + 10),
+            _other => None, // ignore other characters
```

Comment:
I was thinking if folk have an internal representation like 'aaaa-aa-aa' they could render accordingly - perhaps constraint to '_' and a few other chars and ignore rather than leaving it open?